### PR TITLE
[SYCL][Graph] Clarify graph in-order and out-of-order properties

### PR DESCRIPTION
**Pull request by fabiomestre**

### CODE REVIEW

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc [hunk 1]
```diff
@@ -666,6 +666,14 @@ operation as the runtime can perform optimizations based on the graph
 structure. After finalization the graph can be submitted for execution on a
 queue one or more times with reduced overhead.
 
+A `command_graph` can be submitted to both in-order and out-of-order queues. Any
+dependencies between the graph and other command-groups submitted to the same 
+queue will be respected. However, the in-order and out-of-order properties of the
+queue have no effect on how the graph itself is executed (e.g. the graph nodes 
+might execute out-of-order even when using an in-order queue). For further
```
Comment (EwanC) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc]: > how the graph itself is executed

I'd rephrase as "how the nodes within the graph are executed

> (e.g. the graph nodes might execute out-of-order even when using an in-order queue)

I'd also tweak slightly to "e.g. graph nodes without dependency edges may execute out-of-order ...

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc [hunk 2]
```diff
@@ -666,6 +666,14 @@ operation as the runtime can perform optimizations based on the graph
 structure. After finalization the graph can be submitted for execution on a
 queue one or more times with reduced overhead.
 
+A `command_graph` can be submitted to both in-order and out-of-order queues. Any
+dependencies between the graph and other command-groups submitted to the same 
+queue will be respected. However, the in-order and out-of-order properties of the
+queue have no effect on how the graph itself is executed (e.g. the graph nodes 
+might execute out-of-order even when using an in-order queue). For further 
+information about the interaction between queues and graphs
+<<Queue Class Modifications, see the section on Queue Class Modifications>>
```
Comment (EwanC) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc]: It's probably the sub-section on queue properties that's most interesting to link to rather than the whole queue class section.

"For further information about how the properties of a queue affect graphs <<Queue Properties, see the section on Queue Properties>>."

**Review by EwanC — COMMENTED** [review]
In the `syclQueue` parameter description of
```
command_graph(const queue& syclQueue,
              const property_list& propList = {});
```
There's the line "All other properties of the queue are ignored for the purposes of graph creation." Can we add a link there to the Queue Properties section, something like - "See the Queue Properties section for more general information about how queue properties interact with `command_graph` objects."

**Review by Bensuo — APPROVED** [review]

**Comment by fabiomestre** [comment]
> In the `syclQueue` parameter description of
> 
> ```
> command_graph(const queue& syclQueue,
>               const property_list& propList = {});
> ```
> 
> There's the line "All other properties of the queue are ignored for the purposes of graph creation." Can we add a link there to the Queue Properties section, something like - "See the Queue Properties section for more general information about how queue properties interact with `command_graph` objects."

I have added this suggestion. I'm a bit unsure about it though. The details of the queue properties are unrelated to this function (since all of them are ignored). Since this is a description of a parameter which ideally should be succint, it could be crossing the fine line between providing detail and bloating the spec. Don't have a huge objection though.

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc [hunk 3]
```diff
@@ -666,6 +666,14 @@ operation as the runtime can perform optimizations based on the graph
 structure. After finalization the graph can be submitted for execution on a
 queue one or more times with reduced overhead.
 
+A `command_graph` can be submitted to both in-order and out-of-order queues. Any
+dependencies between the graph and other command-groups submitted to the same 
+queue will be respected. However, the in-order and out-of-order properties of the
+queue have no effect on how the nodes within the graph are executed (e.g. the graph
+nodes without dependency edges may execute out-of-order even when using an in-order
+queue). For further information about how the properties of a queue affect graphs
+<<Queue Properties, see the section on Queue Properties>>
```
Comment (EwanC) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc:675]:
```suggestion
<<Queue Properties, see the section on Queue Properties>>.
```

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc [hunk 4]
```diff
@@ -876,7 +884,9 @@ Parameters:
 
 * `syclQueue` - Queue which provides the SYCL device and context for the graph,
   which are immutable characteristics of the graph. All other properties of the
-  queue are ignored for the purposes of graph creation.
+  queue are ignored for the purposes of graph creation. See the
+  <<Queue Properties, Queue Properties>> section for more general information
+  about how queue properties interact with command_graph objects.
```
Comment (EwanC) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc:889]:
```suggestion
  about how queue properties interact with `command_graph` objects.
```

**Review by EwanC — APPROVED** [review]

**Comment by EwanC** [comment]
> I have added this suggestion. I'm a bit unsure about it though. The details of the queue properties are unrelated to this function (since all of them are ignored). Since this is a description of a parameter which ideally should be succint, it could be crossing the fine line between providing detail and bloating the spec. Don't have a huge objection though.

In the email where we got feedback about the in-order/out-of-order properties not being clear, this parameter description of the spec was a place the reader was looking and had questions about. I think the benefit of being able to point to a place with related information in a large spec is worth the extra sentence of verbosity.

**Comment by fabiomestre** [comment]
Upstream PR: https://github.com/intel/llvm/pull/13681
